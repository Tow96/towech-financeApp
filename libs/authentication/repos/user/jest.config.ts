/* eslint-disable */
export default {
  displayName: 'authentication-repos-user',
  preset: '../../../../jest.preset.js',
  testEnvironment: 'node',
  transform: {
    '^.+\\.[tj]s$': ['ts-jest', { tsconfig: '<rootDir>/tsconfig.spec.json' }],
  },
  moduleFileExtensions: ['ts', 'js', 'html'],
  coverageDirectory: '../../../../coverage/libs/authentication/repos/user',
  collectCoverageFrom: [
    '**/*.(t|j)s',
    '!**/*.module.ts',
    '!**/index.ts',
    '!**/jest.config.ts',
    '!**/mocks/**',
  ],
  coverageReporters: ['html', 'text', 'text-summary', 'cobertura', 'lcov', 'json'],
};
