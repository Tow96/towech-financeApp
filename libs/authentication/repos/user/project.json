{
  "name": "authentication-repos-user",
  "$schema": "../../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/authentication/repos/user/src",
  "projectType": "library",
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/authentication/repos/user/**/*.ts"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/authentication/repos/user/jest.config.ts",
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "codeCoverage": true
        }
      }
    }
  },
  "tags": []
}
