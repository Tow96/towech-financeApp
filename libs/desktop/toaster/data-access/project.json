{
  "name": "desktop-toaster-data-access",
  "$schema": "../../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/desktop/toaster/data-access/src",
  "prefix": "towech-finance",
  "tags": [],
  "projectType": "library",
  "targets": {
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/desktop/toaster/data-access/jest.config.ts",
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "codeCoverage": true
        }
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/desktop/toaster/data-access/**/*.ts",
          "libs/desktop/toaster/data-access/**/*.html"
        ]
      }
    }
  }
}
