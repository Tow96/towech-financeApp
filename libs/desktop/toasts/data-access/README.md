# desktop-toaster-data-access

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test desktop-toaster-data-access` to execute the unit tests.
