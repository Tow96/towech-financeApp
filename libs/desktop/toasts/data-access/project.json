{
  "name": "desktop-toasts-data-access",
  "$schema": "../../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/desktop/toasts/data-access/src",
  "prefix": "towech-finance",
  "tags": [],
  "projectType": "library",
  "targets": {
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/desktop/toasts/data-access/jest.config.ts",
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "codeCoverage": true
        }
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/desktop/toasts/data-access/**/*.ts",
          "libs/desktop/toasts/data-access/**/*.html"
        ]
      }
    }
  }
}
