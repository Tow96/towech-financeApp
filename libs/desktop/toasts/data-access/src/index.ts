export * from './lib/desktop-toaster.service';
