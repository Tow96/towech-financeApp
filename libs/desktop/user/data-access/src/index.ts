export * from './lib/desktop-user.service';
