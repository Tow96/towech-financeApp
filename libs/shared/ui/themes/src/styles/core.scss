// The theming functions are heavily taken from angular material, but in a lighter way
// Original code: https://material.angular.io/
@use 'sass:map';

@function get-palette-hue($palette, $hue) {
  @return map.get($palette, $hue);
}

@function get-palette-hue-contrast($palette, $hue) {
  @return map.get(map.get($palette, contrast), $hue);
}

@function define-theme(
  $background,
  $text,
  $primary,
  $accent,
  $success: null,
  $warning: null,
  $error: null
) {
  @if $background == null or $text == null {
    @error 'Background and text colors need to be defined';
  }

  @if $primary == null or $accent == null {
    @error 'primary and accent colors need to be defined palettes';
  }

  @if not map.get($primary, contrast) {
    @error 'Primary is not a palette';
  }

  @if not map.get($accent, contrast) {
    @error 'Accent is not a palette';
  }

  @if $success != null and not map.get($success, contrast) {
    @error 'Success is not a palette';
  }

  @if $error != null and not map.get($error, contrast) {
    @error 'Error is not a palette';
  }

  @if $warning != null and not map.get($warning, contrast) {
    @error 'Warning is not a palette';
  }

  $result: (
    color: (
      text: $text,
      background: $background,
      primary: $primary,
      accent: $accent,
      success: if($success != null, $success, $accent),
      warning: if($warning != null, $warning, $accent),
      error: if($error != null, $error, $accent),
    ),
  );

  @return $result;
}

@mixin color($theme) {
  $colors: map.get($theme, color);
  $primary: map.get($colors, primary);

  :root {
    background-color: map.get($colors, background);
    color: map.get($colors, text);
  }

  div.container {
    background-color: get-palette-hue($primary, 500);
  }
}
