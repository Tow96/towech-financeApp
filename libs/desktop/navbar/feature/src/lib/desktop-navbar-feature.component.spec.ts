// Libraries
import { ComponentFixture, TestBed, tick } from '@angular/core/testing';
import { Router } from '@angular/router';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { subscribeSpyTo } from '@hirez_io/observer-spy';
import { Observable } from 'rxjs';
// Tested elements
import { DesktopNavbarComponent } from './desktop-navbar-feature.component';
// Mocks
import { provideRouter } from '@angular/router';
import { DesktopUserService } from '@towech-finance/desktop/user/data-access';
import { Source } from '@state-adapt/rxjs';

const mockStore = {
  logout$: new Source<void>('aaa'),
};

describe('Desktop Navbar', () => {
  let component: DesktopNavbarComponent;
  let fixture: ComponentFixture<DesktopNavbarComponent>;
  let compiled: HTMLElement;
  let service: DesktopUserService;
  let router: Router;

  beforeEach(() => {
    jest.clearAllMocks();
    TestBed.configureTestingModule({
      imports: [BrowserAnimationsModule],
      providers: [
        { provide: DesktopUserService, useValue: mockStore },
        provideRouter([{ path: 'test', component: DesktopNavbarComponent }]),
      ],
    });
    fixture = TestBed.createComponent(DesktopNavbarComponent);
    component = fixture.componentInstance;

    fixture.detectChanges();
    compiled = fixture.nativeElement;

    service = TestBed.inject(DesktopUserService);
    // router = TestBed.inject(Router);
  });

  it('Must match the snapshot', () => expect(compiled).toMatchSnapshot());

  describe('When logout button is clicked', () => {
    it('Should next a logout event', () => {
      const spy = subscribeSpyTo(service.logout$);

      const logoutBttn = fixture.debugElement.nativeElement.querySelector('#logout > button');
      logoutBttn.click();
      expect(spy.receivedNext()).toBe(true);
    });
  });

  // describe('onToggleCollapse', () => {
  //   it('Should toggle the value of the collapsed variable', () => {
  //     component.collapsed = false;
  //     component.onToggleCollapse();
  //     expect(component.collapsed).toBe(true);
  //     component.onToggleCollapse();
  //     expect(component.collapsed).toBe(false);
  //   });
  // });

  // describe('navigateTo', () => {
  //   it('Should navigate to the correct screen', () => {
  //     jest.clearAllMocks();
  //     const spy = jest.spyOn(router, 'navigate');
  //     component.navigateTo('test');
  //     expect(spy).toHaveBeenCalledTimes(1);
  //     expect(spy).toHaveBeenCalledWith(['test']);
  //   });

  //   it('Should collapse the menu if it is open', () => {
  //     component.collapsed = false;
  //     component.navigateTo('test');
  //     expect(component.collapsed).toBe(true);
  //   });
  // });

  // describe('isRouteActive', () => {
  //   describe('When it is called with the same route as the current', () => {
  //     it('Should return true', () => {
  //       expect(component.isRouteActive('')).toBe(true);
  //     });
  //   });

  //   describe('When it is called with a different route as the current', () => {
  //     it('Should return true', () => {
  //       expect(component.isRouteActive('test')).toBe(false);
  //     });
  //   });
  // });

  // describe('When onOutside click is called', () => {
  //   let clickedTarget: any;

  //   describe('When clicking inside the navbar', () => {
  //     it('Should do nothing', () => {
  //       clickedTarget = compiled.children[0];
  //       component.collapsed = true;
  //       component.clickListener({ target: clickedTarget } as PointerEvent);
  //       expect((component.collapsed = true));

  //       component.collapsed = false;
  //       component.clickListener({ target: clickedTarget } as PointerEvent);
  //       expect((component.collapsed = false));
  //     });
  //   });

  //   describe('When clicking outside the navbar', () => {
  //     it('Should close the navbar', () => {
  //       clickedTarget = document.createElement('p');
  //       component.collapsed = true;
  //       component.clickListener({ target: clickedTarget } as PointerEvent);
  //       expect((component.collapsed = true));

  //       component.collapsed = false;
  //       component.clickListener({ target: clickedTarget } as PointerEvent);
  //       expect((component.collapsed = true));
  //     });
  //   });
  // });

  // describe('When getNavClass is called', () => {
  //   it('Should return the correct classes', () => {
  //     component.collapsed = true;
  //     let result = component.getNavClass();
  //     expect(result).toEqual({
  //       deployed: false,
  //     });

  //     component.collapsed = false;
  //     result = component.getNavClass();
  //     expect(result).toEqual({
  //       deployed: true,
  //     });
  //   });
  // });
});
