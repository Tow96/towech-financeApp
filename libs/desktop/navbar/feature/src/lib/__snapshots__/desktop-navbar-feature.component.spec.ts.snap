// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Desktop Navbar Must match the snapshot 1`] = `
<div
  id="root0"
>
  <div
    class="nav-wrapper"
  >
    <nav>
      <div
        class="header"
      >
        <towech-finance-navbar-item
          icon="bars"
          label="Close"
        >
          <button>
            <div>
              <fa-icon
                class="ng-fa-icon"
                size="2x"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-bars fa-fw fa-2x"
                  data-icon="bars"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
                    fill="currentColor"
                  />
                </svg>
              </fa-icon>
              
            </div>
          </button>
        </towech-finance-navbar-item>
        <div
          class="title"
        >
          <h1>
            Header
          </h1>
        </div>
      </div>
      <div
        class="contents"
      >
        <div
          class="nav__divider desk__only"
        />
        <div
          class="contents__main"
        >
          <towech-finance-navbar-item>
            <button
              disabled=""
            >
              <div>
                <fa-icon
                  class="ng-fa-icon"
                  size="2x"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-money-check-dollar fa-fw fa-2x"
                    data-icon="money-check-dollar"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 576 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M64 64C28.7 64 0 92.7 0 128V384c0 35.3 28.7 64 64 64H512c35.3 0 64-28.7 64-64V128c0-35.3-28.7-64-64-64H64zM272 192H496c8.8 0 16 7.2 16 16s-7.2 16-16 16H272c-8.8 0-16-7.2-16-16s7.2-16 16-16zM256 304c0-8.8 7.2-16 16-16H496c8.8 0 16 7.2 16 16s-7.2 16-16 16H272c-8.8 0-16-7.2-16-16zM164 152v13.9c7.5 1.2 14.6 2.9 21.1 4.7c10.7 2.8 17 13.8 14.2 24.5s-13.8 17-24.5 14.2c-11-2.9-21.6-5-31.2-5.2c-7.9-.1-16 1.8-21.5 5c-4.8 2.8-6.2 5.6-6.2 9.3c0 1.8 .1 3.5 5.3 6.7c6.3 3.8 15.5 6.7 28.3 10.5l.7 .2c11.2 3.4 25.6 7.7 37.1 15c12.9 8.1 24.3 21.3 24.6 41.6c.3 20.9-10.5 36.1-24.8 45c-7.2 4.5-15.2 7.3-23.2 9V360c0 11-9 20-20 20s-20-9-20-20V345.4c-10.3-2.2-20-5.5-28.2-8.4l0 0 0 0c-2.1-.7-4.1-1.4-6.1-2.1c-10.5-3.5-16.1-14.8-12.6-25.3s14.8-16.1 25.3-12.6c2.5 .8 4.9 1.7 7.2 2.4c13.6 4.6 24 8.1 35.1 8.5c8.6 .3 16.5-1.6 21.4-4.7c4.1-2.5 6-5.5 5.9-10.5c0-2.9-.8-5-5.9-8.2c-6.3-4-15.4-6.9-28-10.7l-1.7-.5c-10.9-3.3-24.6-7.4-35.6-14c-12.7-7.7-24.6-20.5-24.7-40.7c-.1-21.1 11.8-35.7 25.8-43.9c6.9-4.1 14.5-6.8 22.2-8.5V152c0-11 9-20 20-20s20 9 20 20z"
                      fill="currentColor"
                    />
                  </svg>
                </fa-icon>
                
              </div>
            </button>
          </towech-finance-navbar-item>
          <towech-finance-navbar-item>
            <button>
              <div>
                <fa-icon
                  class="ng-fa-icon"
                  size="2x"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-gear fa-fw fa-2x"
                    data-icon="gear"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M495.9 166.6c3.2 8.7 .5 18.4-6.4 24.6l-43.3 39.4c1.1 8.3 1.7 16.8 1.7 25.4s-.6 17.1-1.7 25.4l43.3 39.4c6.9 6.2 9.6 15.9 6.4 24.6c-4.4 11.9-9.7 23.3-15.8 34.3l-4.7 8.1c-6.6 11-14 21.4-22.1 31.2c-5.9 7.2-15.7 9.6-24.5 6.8l-55.7-17.7c-13.4 10.3-28.2 18.9-44 25.4l-12.5 57.1c-2 9.1-9 16.3-18.2 17.8c-13.8 2.3-28 3.5-42.5 3.5s-28.7-1.2-42.5-3.5c-9.2-1.5-16.2-8.7-18.2-17.8l-12.5-57.1c-15.8-6.5-30.6-15.1-44-25.4L83.1 425.9c-8.8 2.8-18.6 .3-24.5-6.8c-8.1-9.8-15.5-20.2-22.1-31.2l-4.7-8.1c-6.1-11-11.4-22.4-15.8-34.3c-3.2-8.7-.5-18.4 6.4-24.6l43.3-39.4C64.6 273.1 64 264.6 64 256s.6-17.1 1.7-25.4L22.4 191.2c-6.9-6.2-9.6-15.9-6.4-24.6c4.4-11.9 9.7-23.3 15.8-34.3l4.7-8.1c6.6-11 14-21.4 22.1-31.2c5.9-7.2 15.7-9.6 24.5-6.8l55.7 17.7c13.4-10.3 28.2-18.9 44-25.4l12.5-57.1c2-9.1 9-16.3 18.2-17.8C227.3 1.2 241.5 0 256 0s28.7 1.2 42.5 3.5c9.2 1.5 16.2 8.7 18.2 17.8l12.5 57.1c15.8 6.5 30.6 15.1 44 25.4l55.7-17.7c8.8-2.8 18.6-.3 24.5 6.8c8.1 9.8 15.5 20.2 22.1 31.2l4.7 8.1c6.1 11 11.4 22.4 15.8 34.3zM256 336a80 80 0 1 0 0-160 80 80 0 1 0 0 160z"
                      fill="currentColor"
                    />
                  </svg>
                </fa-icon>
                
              </div>
            </button>
          </towech-finance-navbar-item>
          
        </div>
        <div
          class="nav__divider"
        />
        <div>
          <towech-finance-navbar-item
            id="logout"
            label="Logout"
          >
            <button>
              <div>
                <fa-icon
                  class="ng-fa-icon"
                  size="2x"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-right-from-bracket fa-fw fa-2x"
                    data-icon="right-from-bracket"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M377.9 105.9L500.7 228.7c7.2 7.2 11.3 17.1 11.3 27.3s-4.1 20.1-11.3 27.3L377.9 406.1c-6.4 6.4-15 9.9-24 9.9c-18.7 0-33.9-15.2-33.9-33.9l0-62.1-128 0c-17.7 0-32-14.3-32-32l0-64c0-17.7 14.3-32 32-32l128 0 0-62.1c0-18.7 15.2-33.9 33.9-33.9c9 0 17.6 3.6 24 9.9zM160 96L96 96c-17.7 0-32 14.3-32 32l0 256c0 17.7 14.3 32 32 32l64 0c17.7 0 32 14.3 32 32s-14.3 32-32 32l-64 0c-53 0-96-43-96-96L0 128C0 75 43 32 96 32l64 0c17.7 0 32 14.3 32 32s-14.3 32-32 32z"
                      fill="currentColor"
                    />
                  </svg>
                </fa-icon>
                
              </div>
            </button>
          </towech-finance-navbar-item>
        </div>
      </div>
    </nav>
  </div>
</div>
`;
